pkg/controller/controlplanecerts/controlplanecerts_controller.go

```diff
@@ -380,11 +376,9 @@ func (r *ReconcileControlPlaneCerts) setCertsNotFoundCondition(cd *hivev1.Cluste
 // defaultControlPlaneDomain will attempt to return the domain/hostname for the secondary API URL
 // for the cluster based on the contents of the clusterDeployment's adminKubeConfig secret.
 func (r *ReconcileControlPlaneCerts) defaultControlPlaneDomain(cd *hivev1.ClusterDeployment) (string, error) {
-	remoteClient := r.remoteClientBuilder(cd)
-
-	apiurl, err := remoteClient.UseSecondaryAPIURL().APIURL()
+	apiurl, err := remoteclient.InitialURL(r.Client, cd)
```

Comment:
Ah I think I see, initial is secondary isn't it.

Reply:
Yes. I broke out the `APIURL` function from the `remoteclient.Builder` interface. Consequently, I was no longer beholden to use the terms Primary, Secondary, and Active. Initial API URL is a bit more clear about what the intention is because, when there is no override URL, it is a bit convoluted to say that the secondary is also the initial.